tests/unit/test_session.py

```diff
@@ -45,7 +46,7 @@ def setUp(self):
         self.environ_patch = mock.patch('os.environ', self.environ)
         self.environ_patch.start()
         self.environ['FOO_PROFILE'] = 'foo'
-        self.environ['FOO_REGION'] = 'moon-west-1'
```

Comment:
Interesting.  I assume this is because the region didn't match any of the regexs for the partition right?

To make sure I understand, this would mean a customer would have to upgrade if a new region prefix is added in a partition right?  I'm guessing there's not much we can do to avoid this is there?